### ISSUE TITLE
crash with error

### ISSUE THREAD

**Issue by satman81** [strangelove-ventures/half-life#8]
half life crashed with following error:

panic: sync: unlock of unlocked mutex

goroutine 37244 [running]:
github.com/sasha-s/go-csync.(*Mutex).Unlock(...)
        /root/go/pkg/mod/github.com/sasha-s/go-csync@v0.0.0-20210812194225-61421b77c44b/mutex.go:87
github.com/DisgoOrg/disgo/rest/rrate.(*limiterImpl).Close.func1()
        /root/go/pkg/mod/github.com/!disgo!org/disgo@v0.7.2/rest/rrate/rest_rate_limiter_impl.go:67 +0x10d
created by github.com/DisgoOrg/disgo/rest/rrate.(*limiterImpl).Close
        /root/go/pkg/mod/github.com/!disgo!org/disgo@v0.7.2/rest/rrate/rest_rate_limiter_impl.go:65 +0x94

**Comment by mturkia** [comment]
I am also experiencing this issue. I also seem to have another related panic call trace in the logs. Most of the times the error is as reported in this issue, but the following crash has also occurred several times.
```
panic: sync: unlock of unlocked mutex
goroutine 13 [running]:
github.com/sasha-s/go-csync.(*Mutex).Unlock(...)
        /srv/monitor/go/pkg/mod/github.com/sasha-s/go-csync@v0.0.0-20210812194225-61421b77c44b/mutex.go:87
github.com/DisgoOrg/disgo/rest/rrate.(*limiterImpl).UnlockBucket.func1()
        /srv/monitor/go/pkg/mod/github.com/!disgo!org/disgo@v0.7.2/rest/rrate/rest_rate_limiter_impl.go:149 +0x15f
github.com/DisgoOrg/disgo/rest/rrate.(*limiterImpl).UnlockBucket(0xc0012de7c0, 0xc000d8a000?, 0xc0015ff680)
        /srv/monitor/go/pkg/mod/github.com/!disgo!org/disgo@v0.7.2/rest/rrate/rest_rate_limiter_impl.go:218 +0x7b8
github.com/DisgoOrg/disgo/rest.(*clientImpl).retry(0xc000f42410, 0xc001494140, {0xf211a0?, 0xc001494180?}, {0xde7e20, 0xc0000107d8}, 0x1, {0xc0000107d0, 0x1, 0x1})
        /srv/monitor/go/pkg/mod/github.com/!disgo!org/disgo@v0.7.2/rest/rest_client.go:158 +0xc16
github.com/DisgoOrg/disgo/rest.(*clientImpl).Do(0x0?, 0xc000533d10?, {0xf211a0?, 0xc001494180?}, {0xde7e20?, 0xc0000107d8?}, {0xc0000107d0?, 0x0?, 0x4059000000000000?})
        /srv/monitor/go/pkg/mod/github.com/!disgo!org/disgo@v0.7.2/rest/rest_client.go:201 +0x4f
github.com/DisgoOrg/disgo/rest.(*webhookServiceImpl).UpdateMessage(0xc000d99a10, {0xc00052ca98, 0x12}, {0xc000038690, 0x44}, {0xc00052cb58, 0x12}, {0x0, 0xc000533d10, 0x0,>
        /srv/monitor/go/pkg/mod/github.com/!disgo!org/disgo@v0.7.2/rest/webhook_service.go:179 +0x2b4
github.com/DisgoOrg/disgo/webhook.(*Client).UpdateMessageInThread(0xc0004e2070, {0xc00052cb58?, 0x4059000000000000?}, {0x0, 0xc000533d10, 0x0, 0x0, {0x0, 0x0, 0x0}, ...}, >
        /srv/monitor/go/pkg/mod/github.com/!disgo!org/disgo@v0.7.2/webhook/webhook_client.go:107 +0xe4
github.com/DisgoOrg/disgo/webhook.(*Client).UpdateMessage(...)
        /srv/monitor/go/pkg/mod/github.com/!disgo!org/disgo@v0.7.2/webhook/webhook_client.go:102
github.com/strangelove-ventures/halflife/cmd.(*DiscordNotificationService).UpdateValidatorRealtimeStatus(0xc000ca8fc0, {0x7ffe8e504ef1, 0x18}, 0xc000ca8a80, 0xc000ca6fc0, >
        /srv/monitor/source/half-life/cmd/discord.go:171 +0x3a5
github.com/strangelove-ventures/halflife/cmd.runMonitor({0x123d4e0, 0xc000ca8fc0}, 0xc000cce0a0, 0xc000c1fe18, {0x7ffe8e504ef1, 0x18}, 0x0?, 0xc000ca6fc0, 0x0?)
        /srv/monitor/source/half-life/cmd/validator.go:283 +0xe6
created by github.com/strangelove-ventures/halflife/cmd.glob..func1
        /srv/monitor/source/half-life/cmd/monitor.go:62 +0x614
```

**Comment by agouin** [comment]
Please try `v0.1.7`

**Comment by agouin** [comment]
Closing as resolved in `v0.1.7`
